source/lmp/pair_deepmd.cpp

```diff
@@ -783,6 +784,18 @@ void PairDeepMD::compute(int eflag, int vflag) {
           // Gather std_f and tags
           tagint *tag = atom->tag;
           int nprocs = comm->nprocs;
+          // Grow arrays if necessary
+          if (nlocal > max_nlocal) {
+            max_nlocal = nlocal;
+            memory->destroy(stdfsend);
+            memory->destroy(stdfrecv);
+            memory->destroy(tagsend);
+            memory->destroy(tagrecv);
+            memory->create(stdfsend, nlocal, "deepmd:stdfsendall");
+            memory->create(stdfrecv, nlocal, "deepmd:stdfrecvall");
+            memory->create(tagsend, nlocal, "deepmd:tagsendall");
+            memory->create(tagrecv, nlocal, "deepmd:tagrecvall");
+          }
```

Comment:
btw, I think `all_nlocal` is equal to `atom->natoms`.

Reply:
> `nlocal` is just the number in this processor. Here, for MPI communications, we should use the total number of atoms in all processors, i.e. `all_nlocal`.

I see. `all_nlocal`, summed up from all MPI processes, should be used here. And nothing seems to be wrong since I just used only 1 process to do the test in the last test.